### PR TITLE
CQ-4292656 [Accessibility] Coral.Select include selected value and invalid icon in accessibility name

### PR DESCRIPTION
**Pull request by majornista**
## Description

When setting labelledBy or labelled attribute on Coral.Select, the value or placeholder displayed as the button label, as well as the invalid icon, should be included in the accessibility name for the control.

## Related Issue

https://jira.corp.adobe.com/browse/CQ-4292656

## Motivation and Context

Labelling a Coral.Select control, fails to include the selected value as part of the accessibility name.

## How Has This Been Tested?
Tested use VoiceOver on macOS and NVDA on Windows.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by majornista** [comment]
Please review: @adobre90, @adobe/coral-spectrum, @icaraps.

**Comment by majornista** [comment]
Please review: @adobre90, @adobe/coral-spectrum, @icaraps, @mohiaror

File: coral-component-select/src/scripts/Select.js [hunk 1]
```diff
@@ -482,24 +484,52 @@ class Select extends BaseFormField(BaseComponent(HTMLElement)) {
...
+   */
+  get labelled() {
+    return super.labelled;
+  }
+  set labelled(value) {
+    super.labelled = value;
+  
+    if (this.labelled) {
+      if (!this.labelledBy) {
+        this._elements.button.setAttribute('aria-labelledby', `${this._elements.button.id} ${this._elements.label.id} ${this.invalid ? this._elements.invalidIcon.id : ''}`);
```
Comment (adobre90) [inline coral-component-select/src/scripts/Select.js:499]: What is the point of setting the `aria-label` of button to the id of button?

Comment (majornista) [reply coral-component-select/src/scripts/Select.js:499]: If someone sets label using `labelled` prop, `aria-label` gets added to the `button` element, which will override any of the descendant content. This way, we can label the control using the `button[id][aria-label]`, its `label`, and if necessary, the `invalid` icon.

File: coral-component-select/src/templates/base.html [hunk 2]
```diff
@@ -2,9 +2,9 @@
   var buttonId = data.commons.getUID();
   var listId = data.commons.getUID();
 </js>
-<button tracking="off" variant="_custom" iconposition="right" handle="button" type="button" is="coral-button" aria-haspopup="listbox" id="{{buttonId}}" aria-controls="{{listId}}" aria-expanded="false" class="_coral-FieldButton _coral-Dropdown-trigger">
```
Comment (icaraps) [inline coral-component-select/src/templates/base.html:5]: Why did you remove `type="button"` ?

**Comment by icaraps** [comment]
Can you please add/modify tests ?

File: coral-component-select/src/templates/base.html (hunk 2 above)
Comment (majornista) [reply coral-component-select/src/templates/base.html:5]: That's a mistake. Will fix it.
